File: Landlords/Rest_API/Data/Entities/Landlord.cs
using System.ComponentModel.DataAnnotations;
using FluentValidation;

namespace Rest_API.Data.Entities;

public class Landlord
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public required string Email { get; set; }
    public required string PhoneNumber { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;

    [Required]
    public required string UserId { get; set; }
    public LandlordRestUser User { get; set; }

    public LandlordDto ToDto()
    {
        return new LandlordDto(Id, Name, Email, PhoneNumber, CreatedAt, UpdatedAt);
    }
}

public record LandlordDto(
    int landlordId,
    String name,
    String email,
    String phone_number,
    DateTimeOffset createdAt,
    DateTimeOffset updatedAt
);

public record CreateLandlordDto(String name, String email, String phone_number)
{
    public class CreateLandlordDtoValidator : AbstractValidator<CreateLandlordDto>
    {
        public CreateLandlordDtoValidator()
        {
            RuleFor(x => x.name).NotEmpty().Length(min: 2, max: 50);
            RuleFor(x => x.email).NotEmpty().Length(min: 5, max: 50);
            RuleFor(x => x.phone_number).NotEmpty().Length(min: 8, max: 50);
        }
    }
}

public record UpdateLandlordDto(String name, String email, String phone_number)
{
    public class UpdateLandlordDtoValidatior : AbstractValidator<UpdateLandlordDto>
    {
        public UpdateLandlordDtoValidatior()
        {
            RuleFor(x => x.name).NotEmpty().Length(min: 2, max: 50);
            RuleFor(x => x.email).NotEmpty().Length(min: 5, max: 50);
            RuleFor(x => x.phone_number).NotEmpty().Length(min: 8, max: 50);
        }
    }
};

File: landlords_web_app/lib/backend/landlords_repository/landlord.dart
import 'package:equatable/equatable.dart';

class Landlord extends Equatable {
  final int landlordId;
  final String name;
  final String email;
  final String phoneNumber;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String userId;

  const Landlord({
    required this.landlordId,
    required this.name,
    required this.email,
    required this.phoneNumber,
    required this.createdAt,
    required this.updatedAt,
    required this.userId,
  });

  factory Landlord.fromJson(Map<String, dynamic> json) {
    return Landlord(
      landlordId: json['landlordId'],
      name: json['name'],
      email: json['email'],
      phoneNumber: json['phone_number'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
      userId: json['userId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'landlordId': landlordId,
      'name': name,
      'email': email,
      'phone_number': phoneNumber,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
      'userId': userId,
    };
  }

  Landlord copyWith({
    int? landlordId,
    String? name,
    String? email,
    String? phoneNumber,
    DateTime? createdAt,
    DateTime? updatedAt,
    String? userId,
  }) {
    return Landlord(
      landlordId: landlordId ?? this.landlordId,
      name: name ?? this.name,
      email: email ?? this.email,
      phoneNumber: phoneNumber ?? this.phoneNumber,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      userId: userId ?? this.userId,
    );
  }

  @override
  List<Object?> get props => [
        landlordId,
        name,
        email,
        phoneNumber,
        createdAt,
        updatedAt,
        userId,
      ];
}
